pkg/broker/docker_manager.go

```diff
@@ -0,0 +1,88 @@
+package broker
+
+import (
+	"context"
+
+	"code.cloudfoundry.org/lager"
+	"github.com/cloudfoundry-incubator/blockhead/pkg/config"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/pborman/uuid"
+	"github.com/pivotal-cf/brokerapi"
+)
+
+type dockerContainerManager struct {
+	client DockerClient
+	logger lager.Logger
+}
+
+func NewDockerContainerManager(logger lager.Logger, client DockerClient) ContainerManager {
+	return dockerContainerManager{
+		client: client,
+		logger: logger.Session("docker-container-manager"),
+	}
+}
+
+func (dc dockerContainerManager) Provision(service *config.Service, plan *config.Plan) (brokerapi.ProvisionedServiceSpec, error) {
+	ctx := context.Background()
+	_, err := dc.client.ImagePull(ctx, plan.Image, types.ImagePullOptions{})
+	if err != nil {
+		return brokerapi.ProvisionedServiceSpec{}, err
+	}
+
+	config, hostConfig, netConfig, name := containerCreateOptions(plan)
+	createdContainer, err := dc.client.ContainerCreate(ctx, config, hostConfig, netConfig, name)
+	if err != nil {
+		dc.logger.Error("container-creation-failed", err)
+		return brokerapi.ProvisionedServiceSpec{}, err
+
+	}
+	dc.logger.Info("container-created", lager.Data{"id": createdContainer.ID, "name": name})
+
+	err = dc.client.ContainerStart(ctx, name, types.ContainerStartOptions{})
+	if err != nil {
+		dc.logger.Error("container-start-failed", err)
+		return brokerapi.ProvisionedServiceSpec{}, err
+
+	}
+	dc.logger.Info("container-started", lager.Data{"name": name})
+
+	return brokerapi.ProvisionedServiceSpec{}, nil
+}
+
+func containerCreateOptions(plan *config.Plan) (*container.Config, *container.HostConfig, *network.NetworkingConfig, string) {
+	config := &container.Config{
+		Hostname:     "",
+		Domainname:   "",
+		User:         "",
+		AttachStdin:  false,
+		AttachStdout: true,
+		AttachStderr: true,
+		Tty:          false,
+		OpenStdin:    false,
+		StdinOnce:    false,
```

Comment:
We need to use a field ExposedPorts here so the right ones are opened when the container is created.
https://github.com/moby/moby/blob/master/api/types/container/config.go#L50

Reply:
yup we should get the full list of elements and set them as necessary. I thought I commented `ExposedPorts` out and kept it as placeholder but apparently not.

Reply:
@swetharepakula we got some nice functionality from upstream libs to parse the array of port strings. 
I do not recall if the comment on the above function is good, but it has a lot of functionality in it that lets us do some nice 'magic' to take many different forms of string and still get to the correct end result.
Kube may need similar logic. Not entirely sure.

Reply:
My original comment was about making sure we exposed the ports at this stage.